Video library frontend

UtpalTheDev opened the pull request:
This is my first version of video library. I want suggestions from all of you to make it better. Please review it and give suggestions.

UtpalTheDev committed:
initial commit

RawHeatG commented on frontend/src/styles.css:
```diff
@@ -0,0 +1,546 @@
+  max-height: 100px;
+  overflow: auto;
+}
+.list > li {
+  cursor: pointer;
+  border-bottom: 1px solid #e2e2e2;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  position: relative;
+  padding: 0 0 0 0.3em;
+}
```
Change ``` justify-content: space-between; ``` to ``` justify-content: left; ```
because in Dekstop mode the names get far away

RawHeatG commented on frontend/src/Liked/Liked.js:
```diff
@@ -0,0 +1,87 @@
+  let filteredData = getFilteredData(sortedData, { showCategory });
+  console.log("filtered", filteredData);
+
+  function Likedlist() {
+    return filteredData
+      .filter((item) => item.islike !== false)
+      .map((item) => {
+        return (
```
Add a name or indicator at the top of the page if Videos are there to show the user he's in Liked videos

RawHeatG commented on frontend/src/History/History.js:
```diff
@@ -0,0 +1,93 @@
+  let filteredData = getFilteredData(sortedData, { showCategory });
+  console.log("filtered", filteredData);
+
+  function History() {
+    return filteredData
+      .filter((item) => item.ishistory !== false)
+      .map((item) => {
+        return (
+          <>
+            {" "}
```
Add a name or indicator at the top of page if Videos are there to show the user he's in History videos

RawHeatG reviewed (commented):
The site is Beautiful, and the UX is so good,
Found some small fixes that could be done, please take a look at those

UtpalTheDev committed:
refactor: information about pages not present + style: sortpopup
1. a vast space between sortpopup lists radio button and their titles.

direshpradhan commented on frontend/src/Playlist/Playlist.js:
```diff
@@ -0,0 +1,149 @@
+          {playlist.map((item, index) => {
+            return (
+              <>
+                <li
+                  style={{
+                    background: currentplaylist.id === item.id ? "#f2f2f2" : ""
+                  }}
+                >
```
The background color you have given to differentiate the active playlist is not working, state variable "currentplaylist" is of type number and not an object maybe this is the issue.

direshpradhan commented on frontend/src/Navbar.js:
```diff
@@ -0,0 +1,20 @@
+  return (
+    <>
+      <div className="videonavbar">
+        <div className="videonavbar-logo">
+          <img src={logo} alt="logo" />
+          <span>K</span>rishiGyan
+        </div>
```
When click happens on logo and title it does not route to home page so that can be done which is uniform across all websites.

direshpradhan commented on frontend/src/Reducer-context.js:
```diff
@@ -0,0 +1,313 @@
+      };
+    }
+
+    case "ADD_TO_HISTORY":
+      let historyflag = 0;
+      let historynewarr = state.history.map((item) => {
+        if (item.id === action.payload.id) {
+          historyflag = 1;
+          return {
+            ...item,
+            ishistory: action.payload.ishistory,
+            lastseen: action.payload.lastseen
+          };
+        }
+        return item;
+      });
+      return historyflag === 0
+        ? { ...state, history: [...state.history, action.payload] }
+        : { ...state, history: historynewarr };
```
Instead of using a flag variable, you can use the find method to do the same operation.

direshpradhan commented on frontend/src/video-block.js:
```diff
@@ -0,0 +1,230 @@
+  let { dispatch, playlist } = useReduce();
+  return (
+    <>
+      <div className="playlistmodal">
```
![image](https://user-images.githubusercontent.com/41542536/117571147-d0644480-b0ea-11eb-9b32-c5f4fc16e00b.png)
In the mobile view playlistmodal and the recommended videos are getting overlapped.

direshpradhan reviewed (commented):
Hey Utpal, 
The video library looks great, UI is clean and the features work smoothly. However, you can update some things which will make it better.

UtpalTheDev committed:
refactor: frontend and backend integration

UtpalTheDev committed:
fix: video data not coming from backend

UtpalTheDev committed:
initial commit

added backend

pranshudobhal commented on backend/index.js:
```diff
@@ -0,0 +1,38 @@
+const video=require("./routes/video.router.js")
+
+//mongoose conn
+mongoose.connect('mongodb+srv://utpal:utpalpati@cluster0.pxyfi.mongodb.net/videolib?retryWrites=true&w=majority',{useNewUrlParser: true, useUnifiedTopology: true}).then(()=>{console.log("mongoose connected")}).catch(eror=>{console.log("mongoose connection problem",error)})
```
It is better to put your DB secrets or connection strings in ENV variables

pranshudobhal commented on backend/index.js:
```diff
@@ -0,0 +1,38 @@
+const { errorHandler } = require("./middlewares/error-handler.middleware")
+const { routeNotFound } = require("./middlewares/route-not-found.middleware")
+app.use(cors());
+//const cart = require("./routes/cart.router.js");
```
Unnecessary comments should be removed.

pranshudobhal commented on backend/index.js:
```diff
@@ -0,0 +1,38 @@
+
+app.get("/", (req, res) => {
+  // throw Error("galat")
+  res.send("nahi he")
```
In this, you can send a better response like the name of your API or project.

pranshudobhal commented on backend/models/video.model.js:
```diff
@@ -0,0 +1,16 @@
+
+//schema
+const videoSchema=new mongoose.Schema({
+
```
Unnecessary space added here.

pranshudobhal commented on backend/routes/history.router.js:
```diff
@@ -0,0 +1,75 @@
+const express=require("express");
+const { v4: uuidv4 } = require('uuid');
```
`uuidv4` is not being used in the file. You can remove unused imports.

pranshudobhal commented on backend/routes/history.router.js:
```diff
@@ -0,0 +1,75 @@
+ .get(async (req, res) => {
+   try{
+     const historydata=await historymodel.findOne({historyNo:1});
+     //console.log(history)
```
Here also, you can remove `console.log` comments.

pranshudobhal commented on backend/routes/history.router.js:
```diff
@@ -0,0 +1,75 @@
+   try{
+     const historydata=await historymodel.findOne({historyNo:1});
+     //console.log(history)
+     res.status(200).json({message:"success",historydata:historydata.history})
```
Try to give proper spacing in the JSON response for better readability.

pranshudobhal commented on backend/routes/history.router.js:
```diff
@@ -0,0 +1,75 @@
+  let previtems=await historymodel.findOne({historyNo:1}) ; 
+  if(!previtems){
+    await historymodel.create({historyNo:1,history:[]});
+    let previtems=await historymodel.findOne({historyNo:1}) ; 
```
Since you've already run `findOne` once in `line 25`, there's no need to find it again. 
You can update the `histoyrobj` while creation on new `historymodel` in line 27.

pranshudobhal commented on backend/routes/history.router.js:
```diff
@@ -0,0 +1,75 @@
+  }
+  }
+  await previtems.save();
+  res.json({success:true,product:previtems})
```
There's no use of `product` in the video library. Please check this and change the name accordingly.

pranshudobhal commented on backend/routes/history.router.js:
```diff
@@ -0,0 +1,75 @@
+})
+.delete(async(req,res)=>{
+  try{console.log(req.body)
+  let previtems=await historymodel.findOne({historyNo:1}) ; 
```
Since `historymodel.findOne({historyNo:1})` is being used multiple times, you can write this in a middleware before your API requests.

pranshudobhal commented on backend/routes/liked.router.js:
```diff
@@ -0,0 +1,73 @@
+  }
+  catch (err){
+    console.log(err);
+    res.status(500).json({success:false,message:"unable to add products",errormessage:err.message})
```
You can write better error messages as `unable to add products` doesn't make any sense in the video library.

pranshudobhal reviewed (commented):
Hey @UtpalTheDev,

Your code for the backend looks good! 
Here are few suggestions that might improve your code: 
 - You can use proper indentation in your code so it is easier to read.
 - There are many unnecessary comments and white spaces which can be removed.
 - Your API requests route handlers can be extracted into a separate file for better readability and code maintenance.
 - Your variables are a bit confusing, so it'll be better to give them better names. For eg. If a variable is used to check if a user is present, then it can be named as `isUserPresent`.

I hope this helps! 😀

UtpalTheDev committed:
refactor: reduced the  code length using different logic

UtpalTheDev committed:
fix: liked feature not working properly

UtpalTheDev committed:
refactor: change the logic for history management

UtpalTheDev committed:
fix: delete from history not working

UtpalTheDev committed:
fix: overflow in playlist page, playlist having same item repeatedly

UtpalTheDev committed:
style: flex style change  
1. history, playlist, likedlist element style change.
2. filter list justify content change to start

UtpalTheDev committed:
style: space and padding add, fix: not add to history

UtpalTheDev committed:
style: line clamp add

UtpalTheDev committed:
fix: direct route to video giving error

UtpalTheDev committed:
refactor: remove unuse imports, remove console.log

UtpalTheDev committed:
refactor: remove unuse imports, remove console.log

UtpalTheDev committed:
refactor: auth add

UtpalTheDev committed:
refactor: toast add

UtpalTheDev committed:
refactor: loader add

UtpalTheDev committed:
refactor: 404 page add

UtpalTheDev committed:
refactor: test add

UtpalTheDev committed:
refactor: file structure change

UtpalTheDev committed:
refactor: backend repository change

UtpalTheDev committed:
refactor: title change

UtpalTheDev committed:
refactor: remove unused imports, console logs

UtpalTheDev committed:
fix: image not loading

UtpalTheDev committed:
refactor: padding add

UtpalTheDev committed:
fix: error message not returning

UtpalTheDev committed:
fix: loader not starting

UtpalTheDev committed:
refactor: component library link change

UtpalTheDev committed:
refactor: remove spaces

UtpalTheDev committed:
refactor: manage warnings

UtpalTheDev committed:
Merge pull request #2 from UtpalTheDev/Feature

refactor: manage warnings

UtpalTheDev committed:
refactor: illustration add

UtpalTheDev committed:
refactor:specified width of the illustration

UtpalTheDev committed:
Merge pull request #3 from UtpalTheDev/Feature

Illustration add

UtpalTheDev committed:
refactor: form change

UtpalTheDev committed:
Merge pull request #4 from UtpalTheDev/Feature

refactor: form change

UtpalTheDev committed:
refactor: readme update

UtpalTheDev committed:
Merge pull request #5 from UtpalTheDev/Feature

refactor: readme update

UtpalTheDev committed:
refactor
